utils/useDocsMenu.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useEffect, useState } from 'react'
+import axios, { AxiosResponse } from 'axios'
+import { DocsMenuContent } from '../types/interfaces'
+
+export type DocsMenuHook = {
+  docsMenuContent: DocsMenuContent
+  errorMessage?: string
+}
+
+export default function useDocsMenu(): DocsMenuHook {
```

Comment:
Fine to leave it using Axios for now, but in the future if you're doing data fetching on the client we prefer using `swr` in the `/data` directory.